File: section1/unit3/exercise20.clj
(ns exercise19
  (load-file "list-node.clj"))

(import unit3.list_node.ListNode)

(def f (ListNode. 6 nil))
(def e (ListNode. 5 f))
(def d (ListNode. 4 e))
(def c (ListNode. 3 d))
(def b (ListNode. 2 c))
(def a (ListNode. 1 b))

(defn delete
  [head k]
  (let [result nil tail nil]
    (loop [x 1]
      (when (< x k)
        (let [result (:next head)] (println result))
        (recur (+ x 1))))
    (println result)))


(delete a 3)

File: section1/unit3/exercise20.go
// +build exercise20

package main

import "fmt"

type listNode struct {
	Val  int
	Next *listNode
}

func delete(h *listNode, k int) {
	count := 1
	tmpHead := &listNode{0, h}
	for ; tmpHead.Next != nil; tmpHead = tmpHead.Next {
		if count == k {
			tmpHead.Next = tmpHead.Next.Next
			break
		} else {
			count++
		}
	}
}

func main() {
	f := listNode{6, nil}
	e := listNode{5, &f}
	d := listNode{4, &e}
	c := listNode{3, &d}
	b := listNode{2, &c}
	a := listNode{1, &b}
	delete(&a, 2)
	fmt.Println(a.Next)
}
